rpcserver.go

```diff
@@ -505,6 +505,14 @@ func (r *rpcServer) MintAsset(ctx context.Context,
 	}
 
 	var seedlingMeta *proof.MetaReveal
+
+	// If a custom decimal display is set, the meta type must also be set to
+	// JSON.
+	if req.Asset.DecimalDisplay != 0 && req.Asset.AssetMeta == nil {
+		return nil, fmt.Errorf("decimal display requires JSON asset " +
```

Comment:
Might be better to instead bootstrap the meta json with just the decimal display field? otherwise a user will have to provide an empty json `{}` for this to work, which might not be intuitive

Reply:
I agree with setting to an empty meta json if the meta field nil in the request but a decimal display is given.

Reply:
The `AssetMeta` field here is both `Data` and `Type`. So we require `Type` to be set t JSON for decimal display to get set as the only JSON field if the metadata data field was empty.

Was implemented in https://github.com/lightninglabs/taproot-assets/pull/979 , seen here:

https://github.com/lightninglabs/taproot-assets/blob/311231d67693983b78f48edee1f9fac60efe032d/rpcserver.go#L533

This check is added before that bootstrapping to enforce that the user at least set the correct metadata type.